File: event-service/src/test/java/renovation/event/service/web/controller/StartupControllerTest.java
/*
 * Created under not commercial project "Renovation"
 *
 * Copyright 2021-2025
 */

package renovation.event.service.web.controller;

import org.apache.http.HttpStatus;
import org.hamcrest.CoreMatchers;
import org.junit.jupiter.api.Tag;
import org.junit.jupiter.api.Test;
import org.springframework.test.context.ContextConfiguration;
import renovation.event.service.web.Route;
import renovation.event.service.web.controller.base.KafkaTestcontainersConfigs;
import renovation.event.service.web.controller.base.RestTestInit;

import static renovation.common.util.JsonUtil.simplify;

@Tag("componentTest")
@ContextConfiguration(classes = KafkaTestcontainersConfigs.class)
class StartupControllerTest extends RestTestInit {

    public StartupControllerTest() {
        super(Route.STARTUP);
    }

    @Test
    void about() {
        request
                .when()
                .get("/about")
                .then()
                .statusCode(HttpStatus.SC_OK)
                .body(CoreMatchers.equalTo(
                        simplify(
                                """
                                        {
                                            "name"        : "event-service",
                                            "description" : "Event service module"
                                        }
                                        """)
                ));
    }

    @Test
    void index() {
        request
                .when()
                .get("/module")
                .then()
                .statusCode(HttpStatus.SC_OK)
                .body(CoreMatchers.equalTo("Hi, it's \"Event-service\" module"));
    }
}

File: info/src/test/kotlin/renovation/info/web/controller/StartupControllerTest.kt
/*
 *  Created under not commercial project "Renovation"
 *
 *   Copyright 2021-2023
 */

package renovation.info.web.controller

import io.restassured.module.kotlin.extensions.Then
import io.restassured.module.kotlin.extensions.When
import kotlin.test.Test
import org.apache.http.HttpStatus
import org.hamcrest.CoreMatchers
import org.junit.jupiter.api.Tag
import org.springframework.boot.test.context.SpringBootTest
import org.springframework.boot.test.web.server.LocalServerPort

@Tag("integrationTest")
@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
internal class StartupControllerTest(
    @LocalServerPort val port: Int
) {
    val portHost = "http://localhost:$port"

    @Test
    fun `Assert startup controller`() {
        When {
            get("$portHost/module")
        }.Then {
            statusCode(HttpStatus.SC_OK)
            body(
                CoreMatchers.equalTo("Hi, it's \"Renovation info\" module")
            )
        }
    }

    @Test
    fun `Assert about controller`() {
        When {
            get("$portHost/about")
        }.Then {
            statusCode(HttpStatus.SC_OK)
            body(
                CoreMatchers.equalTo(
                    """{"name":"renovation info module",""" +
                        """"description":"Module work as additional info directory"}"""
                )
            )
        }
    }
}
